packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -0,0 +1,551 @@
+                child: new Stack(
+                  alignment: FractionalOffset.center,
+                  children: <Widget>[
+                    new Align(
+                      alignment: FractionalOffset.topCenter,
+                      child: new Container(
+                        margin: new EdgeInsets.only(
+                          top: new Tween<double>(
+                            begin: 18.0,
+                            end: 6.0
+                          ).evaluate(_animations[i])
+                        ),
+                        child: new IconTheme(
+                          data: new IconThemeData(
+                            color: Colors.white
+                          ),
+                          child: config.labels[i].icon
+                        )
+                      )
+                    ),
+                    new Align(
+                      alignment: FractionalOffset.bottomCenter,
+                      child: new Container(
+                        margin: const EdgeInsets.only(bottom: 10.0),
+                        child: new FadeTransition(
+                          opacity: _animations[i],
+                          child: new DefaultTextStyle(
+                            style: new TextStyle(
+                              fontSize: 14.0,
+                              color: Colors.white
+                            ),
+                            child: config.labels[i].title
+                          )
+                        )
+                      )
+                    )
+                  ]
+                )
+              )
+            )
+          );
+        }
+
+        bottomNavigation = new SizedBox(
+          width: _maxWidth,
+          child: new Row(
+            children: children
+          )
+        );
+
+        break;
+    }
+
+    return new Stack(
+      children: <Widget>[
+        new Positioned.fill(
+          child: new Material(
+            elevation: 8,
+            color: _backgroundColor
+          )
+        ),
+        new SizedBox(
+          height: kBottomNavigationBarHeight,
+          child: new Center(
+            child: new Stack(
+              children: <Widget>[
+                new Positioned(
+                  left: 0.0,
+                  top: 0.0,
+                  right: 0.0,
+                  bottom: 0.0,
+                  child: new CustomPaint(
+                    painter: new _RadialPainter(
+                      circles: _circles.toList()
+                    )
+                  )
+                ),
+                new Material(
+                  type: MaterialType.transparency,
+                  child: new Center(
+                    child: bottomNavigation
+                  )
+                )
+              ]
+            )
+          )
+        )
+      ]
+    );
+  }
+}
+
+class _Circle {
+  _Circle({
+    this.state,
+    this.index,
+    this.color
+  }) {
+    controller = new AnimationController(
```

Comment:
Asserting that the parameters are all non-null here would remind the reader about what's expected.